### ISSUE TITLE
JR renders 105 (Param not allowed) for columns names with underscores in them

### ISSUE THREAD

**Issue by yawboakye** [JSONAPI-Resources/jsonapi-resources#157]
After adding a new column to a model and successfully running the db migration, JR refuses to acknowledge it and accept it as a JSON parameter. Instead it renders the `105 Param not allowed` error. Or am I doing something wrong?

**Comment by lgebhardt** [comment]
Did you add the new attribute to the Resource as well?

**Comment by yawboakye** [comment]
Yes

**Comment by yawboakye** [comment]
@lgebhardt I found out which line is failing: https://github.com/cerebris/jsonapi-resources/blob/master/lib/jsonapi/request.rb#L350. The `include?` test should be made with the formatted key since that's what in `formatted_allowed_fields`.
Also a correction of the original issue: Actually it doesn't fail for newly added columns; it fails for column names separated by underscores.

Here's one way to fix it (PR #158):

``` ruby
params_not_allowed.push(key) unless formatted_allowed_fields.include?(format_key(key).to_sym)
```
